### PR TITLE
Add trigger id to trigger response

### PR DESCRIPTION
**Pull request by benkonz**
adds trigger id to trigger response so that consumers of the trigger endpoint can track a trigger's execution for the /trigger POST api endpoint

### CODE REVIEW

**Comment by benkonz** [comment]
@narape @bergman tomorrow, when you get the chance, could you give this a look?

File: styx-common/src/main/java/com/spotify/styx/model/TriggerResponse.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+package com.spotify.styx.model;
+
+import io.norberg.automatter.AutoMatter;
+
+@AutoMatter
+public interface TriggerResponse extends TriggerRequest {
+  String triggerId();
+
+  static TriggerResponseBuilder builder() {
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/TriggerResponse.java:29]: This is not used outside of this class, so maybe remove it?

File: styx-common/src/main/java/com/spotify/styx/model/TriggerResponse.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return new TriggerResponseBuilder();
+  }
+
+  static TriggerResponse of(WorkflowId workflowId, String parameter,
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/TriggerResponse.java:33]: Maybe a factory method `TriggerResponseBuilder from(TriggerRequest triggerRequest)` could be more useful. A small thing, doesn't matter.

File: styx-scheduler-service/src/test/java/com/spotify/styx/api/SchedulerResourceTest.java [hunk 3]
```diff
@@ -335,13 +336,17 @@ public void testTriggeredWorkflowGeneratesTrigger() throws Exception {
...
+        OBJECT_MAPPER.readValue(response.payload().get().toByteArray(), TriggerResponse.class);
 
     assertThat(response.status(), is(Status.OK));
     final Instant expectedInstant = Instant.parse("2014-12-31T23:00:00.000Z");
     verify(triggerListener).event(eq(HOURLY_WORKFLOW), triggerCaptor.capture(), eq(expectedInstant), eq(expectedParameters));
-    assertThat(TriggerUtil.triggerType(triggerCaptor.getValue()), is("adhoc"));
-    assertThat(TriggerUtil.triggerId(triggerCaptor.getValue()), startsWith("ad-hoc-cli-"));
+    final Trigger trigger = triggerCaptor.getValue();
+    final String triggerId = TriggerUtil.triggerId(trigger);
+    assertThat(TriggerUtil.triggerType(trigger), is("adhoc"));
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/api/SchedulerResourceTest.java:347]: Maybe taking the opportunity to change `assertThat` to `import static org.hamcrest.MatcherAssert.assertThat;`? `org.junit.Assert.assertThat` is deprecated.

**Review by honnix — COMMENTED** [review]
Minor comments. I don't remember how the client and CLI lib work, have you checked whether we need to change anything there?

**Comment by honnix** [comment]
> Minor comments. I don't remember how the client and CLI lib work, have you checked whether we need to change anything there?

NVM I see we don't return anything from client lib.

**Comment by bergman** [comment]
> > Minor comments. I don't remember how the client and CLI lib work, have you checked whether we need to change anything there?
> 
> NVM I see we don't return anything from client lib.

Yeah, I had a look and did not find any, so this will mainly affect any direct integration towards to API. Hope they can accept unknown fields in their deserialisation 😬

File: styx-common/src/main/java/com/spotify/styx/model/TriggerResponse.java (hunk 2 above)
Comment (bergman) [reply styx-common/src/main/java/com/spotify/styx/model/TriggerResponse.java:33]: I think I would rather not leak the builder, but it's not a strong objection 🤷‍♀️

**Review by honnix — APPROVED** [review]

File: styx-scheduler-service/src/test/java/com/spotify/styx/api/SchedulerResourceTest.java (hunk 3 above)
Comment (bergman) [reply styx-scheduler-service/src/test/java/com/spotify/styx/api/SchedulerResourceTest.java:347]: There are ~60 other uses of this import. I fixed this one and can fix the rest in another PR.

Comment (honnix) [reply styx-scheduler-service/src/test/java/com/spotify/styx/api/SchedulerResourceTest.java:347]: Yeah I was only referring the file we anyway need to touch in this PR. The rest can wait.

**Pull request merged**